apps/chat/src/components/AppsEditor/AppsEditorHeader.tsx

```diff
@@ -0,0 +1,201 @@
+import {
+  IconCircleCheck,
+  IconCircleDot,
+  IconMenu2,
+  IconX,
+} from '@tabler/icons-react';
+import { useState } from 'react';
+
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+
+import classNames from 'classnames';
+
+import { useTranslation } from '@/src/hooks/useTranslation';
+
+import { Translation } from '@/src/types/translation';
+
+import { useAppDispatch, useAppSelector } from '@/src/store/hooks';
+import { SettingsSelectors } from '@/src/store/settings/settings.reducers';
+import { UIActions, UISelectors } from '@/src/store/ui/ui.reducers';
+
+import { User } from '@/src/components/Header/User/User';
+import { SettingDialog } from '@/src/components/Settings/SettingDialog';
+
+import LogOutIcon from '../../../public/images/icons/log-out.svg';
+import { Logo } from '../Header/Logo';
+
+export enum TabKeys {
+  GENERAL = 'general',
+  SETTINGS = 'settings',
+}
+
+export const AppsEditorHeader = () => {
+  const dispatch = useAppDispatch();
+  const router = useRouter();
+  const { t } = useTranslation(Translation.Chat);
+
+  const isUserSettingsOpen = useAppSelector(
+    UISelectors.selectIsUserSettingsOpen,
+  );
+  const isOverlay = useAppSelector(SettingsSelectors.selectIsOverlay);
+
+  const handleCloseUserSettings = () => {
+    dispatch(UIActions.setIsUserSettingsOpen(false));
+  };
+
+  const tabs = [
+    {
+      key: TabKeys.GENERAL,
+      label: t('General info'),
+      href: {
+        pathname: `/apps-editor/[slug]`,
+        query: {
+          id: router.query.id?.toString() ?? '',
+          slug: router.query.slug!.toString(),
+        },
+      },
+    },
+    {
+      key: TabKeys.SETTINGS,
+      label: t('Settings'),
+      href: {
+        pathname: `/apps-editor/[slug]/settings`,
+        query: {
+          id: router.query.id?.toString() ?? '',
+          slug: router.query.slug!.toString(),
+        },
+      },
+    },
+  ];
+
+  const [menuOpen, setMenuOpen] = useState(false);
+
+  return (
+    <div
+      className={classNames(
+        'z-40 flex w-full border-b border-tertiary bg-layer-3',
+        isOverlay ? 'min-h-[36px]' : 'min-h-[48px]',
+      )}
+      data-qa="header"
+    >
+      <div className="flex grow items-center justify-between">
+        <div className="flex h-full space-x-4">
+          <div className="flex items-center space-x-4">
+            <button
+              className="p-2 text-primary md:hidden"
+              onClick={() => setMenuOpen(!menuOpen)}
```

Comment:
Updating state like this could lead to unexpected state updates and bugs, because "!menuOpen" may refer to an outdated state value by the time it's actually applied. It is better to use a functional update to guarantee that you receive the most recent state value at the time of the update.
```suggestion
              onClick={() => setMenuOpen((prevState)=> !prevState)}
```